### PR TITLE
feat(Forms): add `showConfirmDialog` to Iterate.RemoveButton

### PR DESCRIPTION
**Pull request by tujoworker**
Based on #4329

Here is a PR [Preview](https://eufemia-git-feat-forms-confirm-remove-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#customize-the-view-and-edit-containers).

### CODE REVIEW

**Comment by langz** [comment]
Pr description says based on #4201, that can't be correct or?🤔

**Comment by langz** [comment]
Tested it quickly on my mobile. Not sure what I think of the order of the buttons. Not that I have any strong opinions, but Inwoild perhaps expect cancel/avbryt to be at the bottom?

image

File: packages/dnb-eufemia/src/extensions/forms/constants/locales/nb-NO.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export default {
     },
     RemoveButton: {
       text: 'Fjern',
+      confirmRemoveText: 'Er du sikker på at du vil slette dette?',
```
Comment (langz) [inline packages/dnb-eufemia/src/extensions/forms/constants/locales/nb-NO.ts:35]:
```suggestion
      confirmRemoveText: 'Er du sikker på at du vil slette?',
```

**Review by langz — APPROVED** [review]

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx [hunk 2]
```diff
@@ -205,7 +205,7 @@ export const ArrayFromFormHandler = () => {
                 </Field.Composition>
 
                 <Iterate.Toolbar>
-                  <Iterate.RemoveButton />
+                  <Iterate.RemoveButton confirmRemove />
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx:208]:
```suggestion
                  <Iterate.RemoveButton confirm />
```

Can it just be confirm? As the component is RemoveButton, so we sort of know the confirming should be about removing. Hence the name of the component.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/RemoveButton/info.mdx [hunk 3]
```diff
@@ -20,3 +20,11 @@ render(
   </>,
 )
 ```
+
+## Confirm removal
+
+You can adding the `confirmRemove` property open a confirmation dialog before removing the item.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/RemoveButton/info.mdx:26]: This sentence could be improved, hmm. Just a suggestion:

```suggestion
You can use the `confirmRemove` property to open a confirmation dialog before removing the item.
```

**Review by langz — APPROVED** [review]

File: packages/dnb-eufemia/src/extensions/forms/constants/locales/nb-NO.ts (hunk 1 above)
Comment (langz) [reply packages/dnb-eufemia/src/extensions/forms/constants/locales/nb-NO.ts:35]: Should we provide an own prop to be able to override this text, or is it good enough providing the possibility to change/override the translation it self?

Comment (tujoworker) [reply packages/dnb-eufemia/src/extensions/forms/constants/locales/nb-NO.ts:35]: I think we add it later when someone requests it.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx (hunk 2 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx]: Yes 👍

Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx]: But on the other hand like more unique props so it's easier to maintain for everyone. 

Could be showConfirmDialog or confirmBeforeRemove

Any thoughts about such prop names?

**Comment by tujoworker** [comment]
> Not entirely sure, but I feel there's something off with the existing order here.

It's a custom toolbar. So apps would need to adjust it to their needs. 
I just used Flex.Horizontal. Changing the order is possible, but would need more code in the example. Which is based on @andlbrei needs.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx (hunk 2 above)
Comment (langz) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/Examples.tsx]: Sure, perhaps just `showConfirmDialog`, I think I like that the best.

**Comment by langz** [comment]
> > Not entirely sure, but I feel there's something off with the existing order here.
> 
> It's a custom toolbar. So apps would need to adjust it to their needs. I just used Flex.Horizontal. Changing the order is possible, but would need more code in the example. Which is based on @andlbrei needs.

I see, then it's fine by me :)

**Comment by langz** [comment]
https://eufemia-git-feat-forms-confirm-remove-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#customize-the-view-and-edit-containers

Some strange animation happening.
The animated item when opening is a bit stuttering, and when closing(avbryt), I'm able to see two items for the one item 🤔 


https://github.com/user-attachments/assets/c70a2836-1fe2-4063-94c0-9ae1e10add38

**Comment by langz** [comment]
In mobile using Safari:

This is probably difficult to resolve. But when removing using the confirm dialog, it sort of laggs/stutters. Sort of feels like it is under heavy load:

https://github.com/user-attachments/assets/c75c7014-72af-4498-8d96-31da224e3efb

**Comment by tujoworker** [comment]
> This is probably difficult to resolve. But when removing using the confirm dialog, it sort of laggs/stutters. Sort of feels like it is under heavy load:

Its not 100% perfect, but I think the latest commit enhances the UX. Let me know your result.

**Comment by tujoworker** [comment]
Before we merge this, we should merge #4329

**Comment by langz** [comment]
> > This is probably difficult to resolve. But when removing using the confirm dialog, it sort of laggs/stutters. Sort of feels like it is under heavy load:
> 
> 
> 
> Its not 100% perfect, but I think the latest commit enhances the UX. Let me know your result.

It felt better now, thanks👏

**Comment by langz** [comment]
> Before we merge this, we should merge #4329

Sure, I let you handle this🫡

**Comment by langz** [comment]
> https://eufemia-git-feat-forms-confirm-remove-eufemia.vercel.app/uilib/extensions/forms/Iterate/Array/demos/#customize-the-view-and-edit-containers
> 
> 
> 
> Some strange animation happening.
> 
> The animated item when opening is a bit stuttering, and when closing(avbryt), I'm able to see two items for the one item 🤔 
> 
> 
> 
> Probably not related to exactly this RemoveButton PR, but yeah.
> 
> 
> 
> https://github.com/user-attachments/assets/c70a2836-1fe2-4063-94c0-9ae1e10add38
> 
> 
> 
> Here's a few screenshots from the video above when opening/closing:
> 
> image
> image

I still experience this. Not sure if this is important to fik, but to me the animation feels a bit strange because it's a few spøit seconds during the animation where I can see both Avatars, as seen in the video:

https://github.com/user-attachments/assets/979b051d-41ff-4db4-b972-abb79123ec24

**Review by langz — APPROVED** [review]

**Pull request merged**